processor/src/lib.rs

```diff
@@ -64,6 +67,18 @@ pub fn execute(script: &Script, inputs: &ProgramInputs) -> Result<ExecutionTrace
     Ok(ExecutionTrace::new(process))
 }
 
+/// Returns an execution iterator that allows callers to step through each exceution and inspect
+/// vm state information along side.
+pub fn execute_iter(
+    script: &Script,
+    inputs: &ProgramInputs,
+) -> Result<VmStateIterator, ExecutionError> {
+    let mut process = Process::new(inputs.clone());
+    process.stack.set_trace_overflow(true);
```

Comment:
Instead of `set_trace_overflow()` method on the `Stack`, I'd prefer to have a separate constructor for `Process` to take an extra parameter for whether we are in debug mode or not, and then pass this parameter to the `Stack` constructor.

This way, we can't instantiate a process in one mode and then switch to another.